sklearn/metrics/_plot/precision_recall.py

```diff
@@ -0,0 +1,172 @@
+
+        self.line_ = ax.plot(self.recall, self.precision, **line_kwargs)[0]
+        ax.set(xlabel="Recall", ylabel="Precision", ylim=[0.0, 1.05],
+               xlim=[0.0, 1.0])
+        ax.legend(loc='lower left')
+
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_precision_recall_curve(estimator, X, y, pos_label=None,
+                                sample_weight=None, response_method="auto",
+                                name=None, ax=None, **kwargs):
+    """Plot Precision Recall Curve.
+
+    Extra keyword arguments will be passed to matplotlib's ``plot``.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Trained classifier.
+
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        Input values.
+
+    y : array-like of shape (n_samples,)
+        Target values.
+
+    pos_label : int or str, default=None
+        The label of the positive class.
+        When `pos_label=None`, if y_true is in {-1, 1} or {0, 1},
+        `pos_label` is set to 1, otherwise an error will be raised.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights.
+
+    response_method : {'predict_proba', 'decision_function', 'auto'} \
+    default='auto'
+        Specifies whether to use :term:`predict_proba` or
+        :term:`decision_function` as the target response. If set to 'auto',
+        :term:`predict_proba` is tried first and if it does not exist
+        :term:`decision_function` is tried next.
+
+    name : str, default=None
+        Name for labeling curve. If `None`, the name of the
+        estimator is used.
+
+    ax : matplotlib axes, default=None
+        Axes object to plot on. If `None`, a new figure and axes is created.
+
+    Returns
+    -------
+    display : :class:`~sklearn.metrics.PrecisionRecallDisplay`
+        Object that stores computed values.
+    """
+    check_matplotlib_support("plot_precision_recall_curve")
+
+    if response_method not in ("predict_proba", "decision_function", "auto"):
+        raise ValueError("response_method must be 'predict_proba', "
+                         "'decision_function' or 'auto'")
+
+    if response_method != "auto":
+        prediction_method = getattr(estimator, response_method, None)
+        if prediction_method is None:
+            raise ValueError(
+                "response method {} is not defined".format(response_method))
+    else:
+        predict_proba = getattr(estimator, 'predict_proba', None)
+        decision_function = getattr(estimator, 'decision_function', None)
+        prediction_method = predict_proba or decision_function
+
+        if prediction_method is None:
+            raise ValueError('response methods not defined')
+
+    y_pred = prediction_method(X)
+
+    if y_pred.ndim != 1:
+        if y_pred.shape[1] > 2:
+            raise ValueError("Estimator should solve a "
```

Comment:
You could fail before to compute the `y_pred` by checking the target `y` using `check_classification_targets`

Reply:
isn't it possible to use `check_classification_targets`?

Reply:
Since this only supports `binary` classification. A `type_of_target(y) == 'binary'` was added before the compute.

Reply:
hm we could also check if the classifier is binary but that might be overkill?
Alternatively we could plot it for a single class even in a multiclass setting by getting the probabilities that correspond to the class we're interested in (that wouldn't work for decision_function necessarily though?). I'm ok with doing this one as a first version, though.